broker/src/test/java/io/moquette/broker/subscriptions/CTrieSpeedTest.java

```diff
@@ -0,0 +1,118 @@
+    public void testFlat() {
+        List<Subscription> results = prepareSubscriptionsFlat();
+        createSubscriptions(results);
+    }
+
+    @Test
+    @Timeout(value = MAX_DURATION_S)
+    public void testDeep() {
+        List<Subscription> results = prepareSubscriptionsDeep();
+        createSubscriptions(results);
+    }
+
+    public void createSubscriptions(List<Subscription> results) {
+        int count = 0;
+        long start = System.currentTimeMillis();
+        int log = CHECK_INTERVAL;
+        CTrie tree = new CTrie();
+        for (Subscription result : results) {
+            tree.addToTree(result);
+            count++;
+            log--;
+            if (log <= 0) {
+                log = CHECK_INTERVAL;
+                long end = System.currentTimeMillis();
+                long duration = end - start;
+                LOGGER.info("Added {} subscriptions in {} ms ({}/ms)", count, duration, Math.round(1.0 * count / duration));
+            }
+            if (Thread.currentThread().isInterrupted()) {
+                return;
+            }
+        }
+        long end = System.currentTimeMillis();
+        long duration = end - start;
+        LOGGER.info("Added " + count + " subscriptions in " + duration + " ms (" + Math.round(1000.0 * count / duration) + "/s)");
+    }
+
+    public List<Subscription> prepareSubscriptionsFlat() {
+        List<Subscription> results = new ArrayList<>(TOTAL_SUBSCRIPTIONS);
+        int count = 0;
+        long start = System.currentTimeMillis();
+        for (int b = 0; b < TOTAL_SUBSCRIPTIONS / 10; b++) {
+            for (int a = 0; a < 10; a++) {
+                count++;
+                results.add(clientSubOnTopic("Client-" + a, "mainTopic-" + b));
+            }
+        }
+        long end = System.currentTimeMillis();
+        long duration = end - start;
+        LOGGER.info("Prepared {} subscriptions in {} ms", count, duration);
+        return results;
+    }
+
+    public List<Subscription> prepareSubscriptionsDeep() {
+        List<Subscription> results = new ArrayList<>(TOTAL_SUBSCRIPTIONS);
+        long countPerLevel = Math.round(Math.pow(TOTAL_SUBSCRIPTIONS, 0.25));
+        LOGGER.info("Preparing {} subscriptions, 4 deep with {} per level", TOTAL_SUBSCRIPTIONS, countPerLevel);
+        int count = 0;
+        long start = System.currentTimeMillis();
+        outerloop:
+        for (int a = 0; a < countPerLevel; a++) {
+            for (int b = 0; b < countPerLevel; b++) {
+                for (int c = 0; c < countPerLevel; c++) {
+                    for (int d = 0; d < countPerLevel; d++) {
+                        count++;
+                        results.add(clientSubOnTopic("Client-" + a, "mainTopic-" + b + "/subTopic-" + c + "/subSubTopic" + d));
+                        if (count >= TOTAL_SUBSCRIPTIONS) {
+                            break outerloop;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
        for (int clientIdx = 0; clientIdx < countPerLevel; clientIdx++) {
            for (int rootLayerIdx = 0; rootLayerIdx < countPerLevel; rootLayerIdx++) {
                for (int firstLayerIdx = 0; firstLayerIdx < countPerLevel; firstLayerIdx++) {
                    for (int secondLayerIdx = 0; secondLayerIdx < countPerLevel; secondLayerIdx++) {
                        count++;
                        results.add(clientSubOnTopic("Client-" + clientIdx, "mainTopic-" + rootLayerIdx + "/subTopic-" + firstLayerIdx + "/subSubTopic" + secondLayerIdx));
                        if (count >= TOTAL_SUBSCRIPTIONS) {
                            break outerloop;
                        }
                    }
                }
            }
        }
```

The same spirit could be applied here, with better naming could increase readability